src/webextension/manage/components/entryDetails.js

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Localized } from "fluent-react/compat";
+import PropTypes from "prop-types";
+import React from "react";
+
+export default class EntryDetails extends React.Component {
+  static get propTypes() {
+    return {
+      entry: PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        site: PropTypes.string.isRequired,
+        username: PropTypes.string.isRequired,
+        password: PropTypes.string.isRequired,
+      }),
+      onSave: PropTypes.func.isRequired,
+      onDelete: PropTypes.func.isRequired,
+    };
+  }
+
+  _setState(props, initial) {
+    let state;
+    if (props.entry)
+      state = {...props.entry};
+    else
+      state = {id: undefined, site: "", username: "", password: ""};
+
+    this.newEntry = !props.entry;
+    if (initial)
+      this.state = state;
+    else
+      this.setState(state);
+  }
+
+  constructor(props) {
+    super(props);
+    this._setState(props, true);
+  }
+
+  componentWillReceiveProps(props) {
+    this._setState(props, false);
+  }
+
+  render() {
+    const {onSave, onDelete} = this.props;
+
+    return (
+      <div style={{margin: "1em"}}>
+        <p>
+          <Localized id="entry-site">
+            <span>sITe</span>
+          </Localized>
+          <input type="text" value={this.state.site}
+                 onChange={(e) => this.setState({site: e.target.value})}/>
+        </p>
+        <p>
+          <Localized id="entry-username">
+            <span>uSERNAMe</span>
+          </Localized>
+          <input type="text" value={this.state.username}
+                 onChange={(e) => this.setState({username: e.target.value})}/>
+        </p>
+        <p>
+          <Localized id="entry-password">
+            <span>pASSWORd</span>
+          </Localized>
+          <input type="text" value={this.state.password}
+                 onChange={(e) => this.setState({password: e.target.value})}/>
+        </p>
+        <p>
+          <Localized id={this.newEntry ? "save-entry" : "update-entry"}>
+            <button onClick={() => onSave(this.state)}>sAVe</button>
```

Comment:
@jimporter I'm curious your thoughts: in order to make <key>Enter</key> submit this non-form form button (vs submit input), we'd need to capture the key press with JS, right? (or potentially change the markup? I'm not sure if React is driving any constraints or requirements here..)

I know you're already thinking ahead on the various accessibility and stuff that needs to happen to polish up I'm just thinking that some of these front end details may be an area I can help pick away at...